### PR TITLE
Convert to json writer following editor data structure

### PR DESCRIPTION
**Pull request by mouthzipper**

### CODE REVIEW

File: lib/writers/json-writer.js [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+                                            text: '',
+                                            marks: [],
+                                            object: 'text'
+                                        }],
+                                        object: 'block'
+                                    }],
+                                })
+                            }
+                        } 
+                        // if(rowspan > 1) {
```
Comment (mmdonaldson) [inline lib/writers/json-writer.js]: Do we need this?

File: lib/writers/json-writer.js [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+var DEFERRED_TYPES = ['th', 'forceWrite'];
+
+exports.writer = writer;
+
+function writer() {
+    return simpleWriter();
+}
+
+function simpleWriter() {
+    function element(node, assetsArray, parentEl) {
```
Comment (mmdonaldson) [inline lib/writers/json-writer.js:15]: Is there a way to maybe split this out into individual functions rather than all the if statements. It's quite hard to follow.

File: lib/writers/json-writer.js [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+        const tagName = getReadableElement(node.tag.tagName);
+        const parenType = parentEl.type;
```
Comment (mmdonaldson) [inline lib/writers/json-writer.js:17]: parentType*

File: lib/writers/json-writer.js (hunk 2 above)
Comment (mouthzipper) [reply lib/writers/json-writer.js:15]: Yes' it's in my todo list already. Planning to break it apart after I get most of the use-case working already.

File: lib/writers/json-writer.js (hunk 1 above)
Comment (mouthzipper) [reply lib/writers/json-writer.js]: Yeah, this is in progress. Logic with rowspan is I need to add new table_row object after the current node, but issue is I need to find current active node with current children iteration after adding it (xN).
